inspirehep/modules/records/receivers.py

```diff
@@ -170,20 +175,106 @@ def populate_bookautocomplete(sender, json, *args, **kwargs):
     input_values.extend(titles)
     input_values = [el for el in input_values if el]
 
-    ref = get_value(json, 'self.$ref')
+    record = get_value(json, 'self.$ref', '')
 
     json.update({
         'bookautocomplete': {
             'input': input_values,
             'payload': {
                 'authors': authors,
-                'id': ref,
+                'id': record,
                 'title': titles,
             },
         },
     })
 
 
+def populate_book_series_suggest(sender, json, *args, **kwargs):
+    """Populate the ``book_series.book_series_suggest`` field of Literature records."""
+    if 'hep.json' not in json.get('$schema'):
+        return
+
+    doc_types = json.get('document_type', [])
+    if all(doc_type not in doc_types for doc_type in ['book', 'thesis', 'proceedings']):
+        return
+
+    book_series = json.get('book_series', [])
+
+    for series in book_series:
+        title = series.get('title')
+        if title:
+            series.update({
+                'book_series_suggest': {
+                    'input': title,
+                    'output': title
+                },
+            })
+
+
+def populate_collaboration_suggest(sender, json, *args, **kwargs):
+    """Populate the ``collaborations.collaboration_suggest`` field of Literature records."""
+    if 'hep.json' not in json.get('$schema'):
+        return
+
+    collaborations = json.get('collaborations', [])
+
+    for collaboration in collaborations:
+        name = collaboration.get('value')
+        if name:
+            collaboration.update({
+                'collaboration_suggest': {
+                    'input': name,
+                    'output': name,
+                },
+            })
+
+
+def populate_conference_suggest(sender, json, *args, **kwargs):
+    """Populate the ``conference_suggest`` field of Conferences records."""
+    if 'conferences.json' not in json.get('$schema'):
+        return
+
+    cnum = json.get('cnum', '')
+    record = get_value(json, 'self.$ref', '')
+
+    json.update({
+        'conference_suggest': {
+            'input': cnum,
```

Comment:
autocompleting on `cnum` is not very useful: either you already know it and you can type it out, or you don't and you won't learn anything from the list of suggestions. You should add more information from the conference record:
`acronyms`, `address`, `series.name`, `titles`.